blkid/blkid_linux_test.go

```diff
@@ -104,19 +105,51 @@ func isoSetup(useJoilet bool) func(t *testing.T, path string) {
 	}
 }
 
-//nolint:gocognit
+func swapSetup(t *testing.T, path string) {
+	t.Helper()
+
+	cmd := exec.Command("mkswap", "--label", "swaplabel", "-p", "8192", path)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	require.NoError(t, cmd.Run())
+}
+
+func swapSetup2(t *testing.T, path string) {
+	t.Helper()
+
+	cmd := exec.Command("mkswap", "--label", "swapswap", "-p", "4096", path)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	require.NoError(t, cmd.Run())
+}
+
+func lvm2Setup(t *testing.T, path string) {
+	t.Helper()
+
+	cmd := exec.Command("pvcreate", "-v", path)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	require.NoError(t, cmd.Run())
+}
+
+//nolint:gocognit,maintidx
 func TestProbePathFilesystems(t *testing.T) {
 	for _, test := range []struct { //nolint:govet
 		name string
 
-		noLoop bool
+		noLoop   bool
+		loopOnly bool
 
 		size  uint64
 		setup func(*testing.T, string)
 
-		expectedName  string
-		expectedLabel string
-		expectUUID    bool
+		expectedName       string
+		expectedLabel      string
+		expectedLabelRegex string
```

Comment:
use a RegExp here and pass in a compiled form: https://pkg.go.dev/regexp#MustCompile